src/DataAggregator/GlobalWorkers/LoopedWorkerBase.cs

```diff
@@ -75,6 +75,72 @@ public enum BehaviourOnFault
     ApplicationExit,
 }
 
+public interface IDelayBetweenLoopsStrategy
+{
+    public static IDelayBetweenLoopsStrategy ConstantDelayStrategy(
+        TimeSpan minDelayAfterSuccess, TimeSpan minDelayAfterError)
+    {
+        // Reusing exponential backoff implementation with an exponent of 0
+        return new ExponentialBackoffDelayBetweenLoopsStrategy(
+            minDelayAfterSuccess, minDelayAfterError,
+            0, 0, 0);
+    }
+
+    TimeSpan DelayAfterSuccess(TimeSpan elapsedSinceLoopBeginning);
+
+    TimeSpan DelayAfterError(TimeSpan elapsedSinceLoopBeginning, uint numConsecutiveErrors);
+}
+
+public class ExponentialBackoffDelayBetweenLoopsStrategy : IDelayBetweenLoopsStrategy
+{
+    private readonly TimeSpan _minDelayAfterSuccess;
+    private readonly TimeSpan _baseDelayAfterError;
+    private readonly int _exponentialBackoffErrorsGracePeriod;
+    private readonly uint _delayAfterErrorExponentialRate;
+    private readonly uint _delayAfterErrorMaxExponent;
+
+    public ExponentialBackoffDelayBetweenLoopsStrategy(
+        TimeSpan minDelayAfterSuccess,
```

Comment:
This is actually `maxDelayAfterSuccess`, or perhaps `delayBetweenLoopTriggersIfSuccessful`